### PR TITLE
[Transform] rename data frame transform to transform for hlrc client

### PR DESCRIPTION
**Pull request by hendrikmuhs**
rename data frame transform to transform for hlrc

### CODE REVIEW

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/docs-check

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/GetTransformResponse.java [hunk 1]
```diff
@@ -42,28 +42,28 @@
...
     }
 
-    private List<DataFrameTransformConfig> transformConfigurations;
+    private List<TransformConfig> transformConfigurations;
     private int count;
     private InvalidTransforms invalidTransforms;
 
-    public GetDataFrameTransformResponse(List<DataFrameTransformConfig> transformConfigurations,
+    public GetTransformResponse(List<TransformConfig> transformConfigurations,
                                          int count,
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/GetTransformResponse.java]: #nit indention fix.
```suggestion
                                int count,
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/GetTransformResponse.java [hunk 2]
```diff
@@ -42,28 +42,28 @@
...
                                          @Nullable InvalidTransforms invalidTransforms) {
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/GetTransformResponse.java]:
```suggestion
                                @Nullable InvalidTransforms invalidTransforms) {
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/StopTransformResponse.java [hunk 3]
```diff
@@ -28,18 +28,18 @@
...
+            .generateParser("stop_transform_response", StopTransformResponse::new, ACKNOWLEDGED);
 
-    public static StopDataFrameTransformResponse fromXContent(final XContentParser parser) throws IOException {
+    public static StopTransformResponse fromXContent(final XContentParser parser) throws IOException {
         return PARSER.parse(parser, null);
     }
 
-    public StopDataFrameTransformResponse(boolean acknowledged, @Nullable List<TaskOperationFailure> taskFailures,
+    public StopTransformResponse(boolean acknowledged, @Nullable List<TaskOperationFailure> taskFailures,
                                           @Nullable List<? extends ElasticsearchException> nodeFailures) {
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/StopTransformResponse.java]:
```suggestion
                                 @Nullable List<? extends ElasticsearchException> nodeFailures) {
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformCheckpointStats.java [hunk 4]
```diff
@@ -28,47 +28,47 @@
...
-    public static DataFrameTransformCheckpointStats fromXContent(XContentParser parser) throws IOException {
+    public static TransformCheckpointStats fromXContent(XContentParser parser) throws IOException {
         return LENIENT_PARSER.parse(parser, null);
     }
 
-    public DataFrameTransformCheckpointStats(final long checkpoint, final DataFrameIndexerPosition position,
-                                             final DataFrameTransformProgress checkpointProgress, final long timestampMillis,
+    public TransformCheckpointStats(final long checkpoint, final TransformIndexerPosition position,
+                                             final TransformProgress checkpointProgress, final long timestampMillis,
                                              final long timeUpperBoundMillis) {
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformCheckpointStats.java]:
```suggestion
                                    final long timeUpperBoundMillis) {
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformCheckpointingInfo.java [hunk 5]
```diff
@@ -29,46 +29,46 @@
...
         LENIENT_PARSER.declareField(ConstructingObjectParser.optionalConstructorArg(),
             p -> TimeUtil.parseTimeFieldToInstant(p, CHANGES_LAST_DETECTED_AT.getPreferredName()),
             CHANGES_LAST_DETECTED_AT,
             ObjectParser.ValueType.VALUE);
     }
 
-    public DataFrameTransformCheckpointingInfo(DataFrameTransformCheckpointStats last,
-                                               DataFrameTransformCheckpointStats next,
+    public TransformCheckpointingInfo(TransformCheckpointStats last,
+                                               TransformCheckpointStats next,
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformCheckpointingInfo.java]:
```suggestion
                                      TransformCheckpointStats next,
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformCheckpointingInfo.java [hunk 6]
```diff
@@ -29,46 +29,46 @@
...
                                                long operationsBehind,
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformCheckpointingInfo.java]:
```suggestion
                                      long operationsBehind,
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformCheckpointingInfo.java [hunk 7]
```diff
@@ -29,46 +29,46 @@
...
                                                Instant changesLastDetectedAt) {
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformCheckpointingInfo.java]:
```suggestion
                                      Instant changesLastDetectedAt) {
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformConfig.java [hunk 8]
```diff
@@ -109,26 +109,26 @@ private static SyncConfig parseSyncConfig(XContentParser parser) throws IOExcept
...
+     * @return A TransformConfig to preview, NOTE it will have a {@code null} id, destination and index.
      */
-    public static DataFrameTransformConfig forPreview(final SourceConfig source, final PivotConfig pivotConfig) {
-        return new DataFrameTransformConfig(null, source, null, null, null, pivotConfig, null, null, null);
+    public static TransformConfig forPreview(final SourceConfig source, final PivotConfig pivotConfig) {
+        return new TransformConfig(null, source, null, null, null, pivotConfig, null, null, null);
     }
 
-    DataFrameTransformConfig(final String id,
+    TransformConfig(final String id,
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformConfig.java]: indention for all these params...I cannot provide the suggestion to LOC outside of the change :(((((

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformConfigUpdate.java [hunk 9]
```diff
@@ -74,7 +74,7 @@ private static SyncConfig parseSyncConfig(XContentParser parser) throws IOExcept
     private final SyncConfig syncConfig;
     private final String description;
 
-    public DataFrameTransformConfigUpdate(final SourceConfig source,
+    public TransformConfigUpdate(final SourceConfig source,
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformConfigUpdate.java]: #nit indention

File: client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformStats.java [hunk 10]
```diff
@@ -40,35 +40,35 @@
...
     private final NodeAttributes node;
-    private final DataFrameIndexerTransformStats indexerStats;
-    private final DataFrameTransformCheckpointingInfo checkpointingInfo;
+    private final TransformIndexerStats indexerStats;
+    private final TransformCheckpointingInfo checkpointingInfo;
 
-    public DataFrameTransformStats(String id, State state, String reason, NodeAttributes node, DataFrameIndexerTransformStats stats,
-                                   DataFrameTransformCheckpointingInfo checkpointingInfo) {
+    public TransformStats(String id, State state, String reason, NodeAttributes node, TransformIndexerStats stats,
+                                   TransformCheckpointingInfo checkpointingInfo) {
```
Comment (benwtrent) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformStats.java]:
```suggestion
                          TransformCheckpointingInfo checkpointingInfo) {
```

File: x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java [hunk 11]
```diff
@@ -195,21 +195,21 @@ protected PivotConfig createPivotConfig(Map<String, SingleGroupSource> groups,
         return builder.build();
     }
 
-    protected DataFrameTransformConfig createTransformConfig(String id,
+    protected TransformConfig createTransformConfig(String id,
                                                              Map<String, SingleGroupSource> groups,
```
Comment (benwtrent) [inline x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java]:
```suggestion
                                                    Map<String, SingleGroupSource> groups,
```

File: x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java [hunk 12]
```diff
@@ -195,21 +195,21 @@ protected PivotConfig createPivotConfig(Map<String, SingleGroupSource> groups,
...
                                                              AggregatorFactories.Builder aggregations,
```
Comment (benwtrent) [inline x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java]:
```suggestion
                                                    AggregatorFactories.Builder aggregations,
```

File: x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java [hunk 13]
```diff
@@ -195,21 +195,21 @@ protected PivotConfig createPivotConfig(Map<String, SingleGroupSource> groups,
...
                                                              String destinationIndex,
```
Comment (benwtrent) [inline x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java]:
```suggestion
                                                    String destinationIndex,
```

File: x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java [hunk 14]
```diff
@@ -195,21 +195,21 @@ protected PivotConfig createPivotConfig(Map<String, SingleGroupSource> groups,
...
                                                              String... sourceIndices) throws Exception {
```
Comment (benwtrent) [inline x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java]:
```suggestion
                                                    String... sourceIndices) throws Exception {
```

File: x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java [hunk 15]
```diff
@@ -195,21 +195,21 @@ protected PivotConfig createPivotConfig(Map<String, SingleGroupSource> groups,
...
         return createTransformConfig(id, groups, aggregations, destinationIndex, QueryBuilders.matchAllQuery(), sourceIndices);
     }
 
-    protected DataFrameTransformConfig.Builder createTransformConfigBuilder(String id,
+    protected TransformConfig.Builder createTransformConfigBuilder(String id,
                                                                             Map<String, SingleGroupSource> groups,
```
Comment (benwtrent) [inline x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java]:
```suggestion
                                                                   Map<String, SingleGroupSource> groups,
```

File: x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java [hunk 16]
```diff
@@ -195,21 +195,21 @@ protected PivotConfig createPivotConfig(Map<String, SingleGroupSource> groups,
...
                                                                             AggregatorFactories.Builder aggregations,
```
Comment (benwtrent) [inline x-pack/plugin/transform/qa/multi-node-tests/src/test/java/org/elasticsearch/xpack/transform/integration/TransformIntegTestCase.java]:
```suggestion
                                                                   AggregatorFactories.Builder aggregations,
```

**Review by benwtrent — APPROVED** [review]
Minor cosmetic things. Just some indention woes given the rename :)

`:%s/data.*frame//gi`

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/bwc

**Pull request merged**